src/ui/page_storage.c

```diff
@@ -584,6 +584,10 @@ static void *page_storage_repair_thread(void *arg) {
         page_storage.is_auto_sd_repair_active = false;
     }
     page_storage.is_sd_repair_complete = true;
+
+    if (complete_callback != NULL) {
+        complete_callback();
```

Comment:
The way complete_callback() is currently implemented can potentially blow up the stack for this given thread since the calling thread is not exited until all the nested callbacks have completed.